flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/expressions/RexNodeConverter.java

```diff
@@ -107,94 +341,295 @@ public RexNode visit(CallExpression call) {
 		}
 	}
 
+	private RexNode convertOver(List<Expression> children) {
+		List<Expression> args = children;
+		Expression agg = args.get(0);
+		SqlAggFunction aggFunc = agg.accept(new SqlAggFunctionVisitor(typeFactory));
+		RelDataType aggResultType = typeFactory.createFieldTypeFromLogicalType(
+				fromDataTypeToLogicalType(((ResolvedExpression) agg).getOutputDataType()));
+
+		// assemble exprs by agg children
+		List<RexNode> aggExprs = agg.getChildren().stream().map(expr -> expr.accept(this))
+				.collect(Collectors.toList());
+
+		// assemble order by key
+		Expression orderKeyExpr = args.get(1);
+		Set<SqlKind> kinds = new HashSet<>();
+		RexNode collationRexNode = createCollation(orderKeyExpr.accept(this), RelFieldCollation.Direction.ASCENDING,
+				null, kinds);
+		ImmutableList<RexFieldCollation> orderKey = ImmutableList
+				.of(new RexFieldCollation(collationRexNode, kinds));
+
+		// assemble partition by keys
+		List<RexNode> partitionKeys = args.subList(4, args.size()).stream().map(expr -> expr.accept(this))
+				.collect(Collectors.toList());
+		// assemble bounds
+		Expression preceding = args.get(2);
+		boolean isPhysical = ((ResolvedExpression) preceding).getOutputDataType().equals(DataTypes.BIGINT());
+		Expression following = args.get(3);
+		RexWindowBound lowerBound = createBound(preceding, SqlKind.PRECEDING);
+		RexWindowBound upperBound = createBound(following, SqlKind.FOLLOWING);
+
+		// build RexOver
+		return relBuilder.getRexBuilder().makeOver(
+				aggResultType,
+				aggFunc,
+				aggExprs,
+				partitionKeys,
+				orderKey,
+				lowerBound,
+				upperBound,
+				isPhysical,
+				true,
+				false,
+				false);
+	}
+
+	private RexNode convertAs(List<Expression> children) {
+		String name = extractValue((ValueLiteralExpression) children.get(1), String.class);
+		RexNode child = children.get(0).accept(this);
+		return relBuilder.alias(child, name);
+	}
+
+	private RexNode convertTrim(List<Expression> children) {
+		ValueLiteralExpression removeLeadingExpr = (ValueLiteralExpression) children.get(0);
+		Boolean removeLeading = extractValue(removeLeadingExpr, Boolean.class);
+		ValueLiteralExpression removeTrailingExpr = (ValueLiteralExpression) children.get(1);
+		Boolean removeTrailing = extractValue(removeTrailingExpr, Boolean.class);
+		RexNode trimString = children.get(2).accept(this);
+		RexNode str = children.get(3).accept(this);
+		Enum trimMode;
+		if (removeLeading && removeTrailing) {
+			trimMode = SqlTrimFunction.Flag.BOTH;
+		} else if (removeLeading) {
+			trimMode = SqlTrimFunction.Flag.LEADING;
+		} else if (removeTrailing) {
+			trimMode = SqlTrimFunction.Flag.TRAILING;
+		} else {
+			throw new IllegalArgumentException("Unsupported trim mode.");
+		}
+		return relBuilder.call(
+				FlinkSqlOperatorTable.TRIM,
+				relBuilder.getRexBuilder().makeFlag(trimMode),
+				trimString,
+				str);
+	}
+
+	private RexNode ConvertGet(List<Expression> children) {
```

Comment:
code style error..

Reply:
sorry for that, @JingsongLi fixed the compile error in #8689